url04f7fcf4e1_peteratt_HDFS_Replication_Simulator_tgz-pJ8-hdfs_replicationsimulator_MainJ8/src/hdfs/replicationsimulator/NodeKiller.java

```diff
@@ -0,0 +1,29 @@
+package hdfs.replicationsimulator;
+
+import java.util.List;
+
+public class NodeKiller implements Runnable {
+
+	@Override
+	public void run() {
+		
+		Simulator.addTrace(new SimTrace("Started NodeKiller"));
+		
+		long initialTime = 0;
+		List<Event> failures = Simulator.getSimulationFailureEvents();
+		
+		for (Event e : failures) {
+			
+			try {
+				Thread.sleep(e.getTime() - initialTime);
+				initialTime = e.getTime();
+			} catch (InterruptedException e1) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)